src/userSurvey.js

```diff
@@ -162,7 +173,21 @@ export async function setSurvey(options) {
         "webScience.userSurvey.popupNoPromptMessage": options.popupNoPromptMessage
     });
 
-    storageSpace = storage.createKeyValueStorage("webScience.userSurvey");
+    initializeStorage();
+
+    let currentSurvey = await storageSpace.get("currentSurvey");
+
+    /* Clears the existing survey data if this survey differs from the
+     * previous survey.
+     */
+    if (currentSurvey !== options.surveyName) {
+        await storageSpace.set("lastSurveyRequest", null);
+        await storageSpace.set("surveyCompleted", false);
+        await storageSpace.set("surveyCancelled", false);
+    }
```

Comment:
Also, the current implementation seems bugged... `createdSurvey` will be `true` after the first call to `setSurvey`. A subsequent call will result in an error because of that. There would have to be an extension reload between calls of `setSurvey`.

Reply:
Created `endCurrentSurvey` function that:

1. Clears the existing survey reminder timeout, removes the listener
2. Removes the `browser.webRequest.onBeforeRequest` listener that checks for survey completion
3. Clears the values of `lastSurveyRequest`, `surveyCompleted`, `surveyCancelled`, and `currentSurvey` in storage.